src/utils/useMint.tsx

```diff
@@ -0,0 +1,192 @@
+import { useWallet } from "@mintbase-js/react";
+import { uploadReference } from "@mintbase-js/storage";
+import { useState } from "react";
+import { convertBase64ToFile } from "./base64ToFile";
+import { generateRandomId } from "./generateRandomId";
+import { useReplicate } from "@/providers/replicate";
+import { constants } from "@/constants";
+
+interface ReferenceObject {
+  title: string;
+  description: string;
+  media: File;
+}
+
+const useMintImage = () => {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const { selector, activeAccountId } = useWallet();
+  const { addRequest } = useReplicate();
+
+  const getWallet = async () => {
+    try {
+      return await selector.wallet();
+    } catch (error) {
+      console.error("Failed to retrieve the wallet:", error);
+      setLoading(false);
+      throw new Error("Failed to retrieve the wallet");
+    }
+  };
+
+  const getTitleAndDescription = async (photo: string) => {
+    try {
+      const requestPayload = {
+        image: photo,
+        prompt: `Describe this image, be direct and include important details. The title should be succinct and 5 words long. The description can be longer than 15 words and more descriptive.
+      
+      Respond in JSON {"title": "<title>", "description": "<description>"}`,
+      };
+      const requestHash =
+        "2facb4a474a0462c15041b78b1ad70952ea46b5ec6ad29583c0b29dbd4249591";
+      const response = await addRequest(requestPayload, requestHash);
+
+      return JSON.parse(response.output.join(""));
+    } catch (error) {
+      console.error("Failed to get title and description:", error);
+      return {
+        title: generateRandomId(10),
+        description: generateRandomId(10),
+      };
+    }
+  };
+
+  const uploadReferenceObject = async (refObject: ReferenceObject) => {
+    try {
+      return await uploadReference(refObject);
+    } catch (error) {
+      console.error("Failed to upload reference:", error);
+      setLoading(false);
+      throw new Error("Failed to upload reference");
+    }
+  };
+
+  const performTransaction = async (
+    wallet: any,
+    metadata: any,
+    baseUrl: string
+  ) => {
+    if (!wallet) {
+      throw new Error("Wallet is not defined.");
+    }
+
+    try {
+      return await wallet.signAndSendTransaction({
+        signerId: activeAccountId,
+        receiverId: constants.proxyContractAddress,
+        actions: [
+          {
+            type: "FunctionCall",
+            params: {
+              methodName: "mint",
+              args: {
+                metadata: JSON.stringify(metadata),
+                nft_contract_id: constants.tokenContractAddress,
+              },
+              gas: "200000000000000",
+              deposit: "10000000000000000000000",
+            },
+          },
+        ],
+        callbackUrl: baseUrl,
```

Comment:
this wont be needed as pass the callbackUrl globally on the provider on #52 


check 
https://github.com/Mintbase/minsta/blob/db09c758503bbb9572db861da3ae1904838907b4/src/providers/providers.tsx

Reply:
https://github.com/Mintbase/minsta/blob/4d978c82edea2554e7c2a366026f15f8d492bb49/src/providers/app.tsx#L125

This is in prod right now too. I've refactored based on this file alone, shall I update this accordingly?